pkg/archive/archive_test.go

```diff
@@ -207,6 +207,164 @@ func TestCmdStreamGood(t *testing.T) {
 	}
 }
 
+func TestUntarPathWithInvalidSrc(t *testing.T) {
+	_, err := os.Create("/tmp/dest")
+	if err != nil {
+		t.Fatalf("Fail to create the destination file")
+	}
+	defer os.RemoveAll("/tmp/dest")
+	err = UntarPath("", "/tmp/dest")
+	if err == nil {
+		t.Fatalf("UntarPath with invalid src path should throw an error.")
+	}
+}
+
+func TestUntarPath(t *testing.T) {
+	cmd := exec.Command("/bin/sh", "-c", "touch /tmp/src && tar cfT /tmp/src.tar /tmp/src")
```

Comment:
I would recommend that you create a new/temp dir for stuff like this otherwise you may end up with a conflict with pre-existing files on the system.